File: unit/unit/src/main/java/com/cedarsoft/unit/si/microM.java
package com.cedarsoft.unit.si;

import java.lang.annotation.Documented;
import java.lang.annotation.Inherited;

import com.cedarsoft.unit.Name;
import com.cedarsoft.unit.Symbol;
import com.cedarsoft.unit.Unit;
import com.cedarsoft.unit.prefix.micro;
import com.cedarsoft.unit.quantity.Length;

/**
 * Micrometer
 */
@Documented
@Inherited

@Unit
@Length
@Symbol(microM.SYMBOL)
@Name("micrometer")
@micro(m.class)

public @interface microM {
  String SYMBOL = "μm";
}

File: unit/unit/src/commonMain/kotlin/com/cedarsoft/unit/si/microM.kt
package com.cedarsoft.unit.si


import com.cedarsoft.unit.Name
import com.cedarsoft.unit.Symbol
import com.cedarsoft.unit.Unit
import com.cedarsoft.unit.prefix.micro
import com.cedarsoft.unit.quantity.Length

/**
 * Micrometer
 */
@Retention(AnnotationRetention.SOURCE)
@Target(
  AnnotationTarget.CLASS,
  AnnotationTarget.ANNOTATION_CLASS,
  AnnotationTarget.TYPE_PARAMETER,
  AnnotationTarget.PROPERTY,
  AnnotationTarget.FIELD,
  AnnotationTarget.LOCAL_VARIABLE,
  AnnotationTarget.VALUE_PARAMETER,
  AnnotationTarget.CONSTRUCTOR,
  AnnotationTarget.FUNCTION,
  AnnotationTarget.PROPERTY_GETTER,
  AnnotationTarget.PROPERTY_SETTER,
  AnnotationTarget.TYPE,
  AnnotationTarget.EXPRESSION,
  AnnotationTarget.FILE,
  AnnotationTarget.TYPEALIAS
)
@MustBeDocumented
@Unit
@Length
@Symbol(microM.SYMBOL)
@Name("micrometer")
@micro(m::class)
annotation class microM {
  companion object {
    const val SYMBOL: String = "μm"
  }
}
